src/hotspot/cpu/riscv/templateTable_riscv_64.cpp

```diff
@@ -973,40 +975,40 @@ void TemplateTable::dastore() {
 
 // Pop 3 values from the stack and...
 void TemplateTable::aastore() {
+  tty->print_cr("aastore: %p", __ pc());
   transition(vtos, vtos);
 
   Label Lstore_ok, Lis_null, Ldone;
-  const Register Rindex    = R3_ARG1_PPC,
-                 Rarray    = R4_ARG2_PPC,
+  const Register Rindex    = R10_ARG0,
+                 Rarray    = R11_ARG1,
                  Rscratch  = R5_scratch1,
                  Rscratch2 = R6_scratch2,
-                 Rarray_klass = R5_ARG3_PPC,
+                 Rarray_klass = R12_ARG2,
                  Rarray_element_klass = Rarray_klass,
-                 Rvalue_klass = R6_ARG4_PPC,
-                 Rstore_addr = R31;    // Use register which survives VM call.
+                 Rvalue_klass = R13_ARG3,
+                 Rstore_addr = R14_ARG4;    // Use register which survives VM call.
 
-  __ ld_PPC(R25_tos, Interpreter::expr_offset_in_bytes(0), R23_esp); // Get value to store.
-  __ lwz_PPC(Rindex, Interpreter::expr_offset_in_bytes(1), R23_esp); // Get index.
-  __ ld_PPC(Rarray, Interpreter::expr_offset_in_bytes(2), R23_esp);  // Get array.
+  __ ld(R25_tos, R23_esp, Interpreter::expr_offset_in_bytes(1)); // Get value to store.
+  __ lwu(Rindex, R23_esp, Interpreter::expr_offset_in_bytes(2)); // Get index.
+  __ ld(Rarray, R23_esp, Interpreter::expr_offset_in_bytes(3));  // Get array.
 
   __ verify_oop(R25_tos);
   __ index_check_without_pop(Rarray, Rindex, UseCompressedOops ? 2 : LogBytesPerWord, Rscratch, Rstore_addr);
   // Rindex is dead!
   Register Rscratch3 = Rindex;
 
   // Do array store check - check for NULL value first.
-  __ cmpdi_PPC(CCR0, R25_tos, 0);
-  __ beq_PPC(CCR0, Lis_null);
+  __ beqz(R25_tos, Lis_null);
 
   __ load_klass(Rarray_klass, Rarray);
   __ load_klass(Rvalue_klass, R25_tos);
 
   // Do fast instanceof cache test.
-  __ ld_PPC(Rarray_element_klass, in_bytes(ObjArrayKlass::element_klass_offset()), Rarray_klass);
+  __ ld(Rarray_element_klass, Rarray_klass, in_bytes(ObjArrayKlass::element_klass_offset()));
 
   // Generate a fast subtype check. Branch to store_ok if no failure. Throw if failure.
   // FIXME_RISCV this call falls in different registers assert
-  //  __ gen_subtype_check(Rvalue_klass /*subklass*/, Rarray_element_klass /*superklass*/, Rscratch, Rscratch2, Rscratch3, Lstore_ok);
+  __ j(Lstore_ok); // __ gen_subtype_check(Rvalue_klass /*subklass*/, Rarray_element_klass /*superklass*/, Rscratch, Rscratch2, Rscratch3, Lstore_ok);
```

Comment:
Type check is important for java. Is it hard to implement this or at least make a verbose logging it is not performed?